retis/src/core/filters/meta/filter.rs

```diff
@@ -268,160 +569,558 @@ impl MetaOp {
             };
         }
 
-        if mask > 0 {
-            if lop.is_ptr() || (lop.is_num() && !lop.is_signed()) {
-                lop.mask = mask;
+        if field.mask > 0 {
+            if ctx.load.is_ptr() || (ctx.load.is_num() && !ctx.load.is_signed()) {
+                ctx.load.mask = field.mask;
             } else {
                 bail!("mask is only supported for pointers and unsigned numeric members.");
             }
         }
 
-        lop.bf_size = u8::try_from(bfs)?;
-        lop.offt = u16::try_from(offt)?;
-
-        if bfs == 0 {
-            lop.offt /= 8;
-        }
-
-        Ok(op)
-    }
+        ctx.load.offt = u16::try_from(self.offt)?;
 
-    fn emit_target(lmo: &MetaLoad, rval: Rval, cmp_op: MetaCmp) -> Result<MetaOp> {
-        let mut op: MetaOp = MetaOp::new();
-        let top = op.target_ref_mut();
-
-        if lmo.is_ptr() || lmo.nmemb > 0 {
-            if cmp_op != MetaCmp::Eq && cmp_op != MetaCmp::Ne {
+        if ctx.load.is_ptr() || ctx.load.nmemb > 0 {
+            if rel_op != RelOp::Eq && rel_op != RelOp::Ne {
                 bail!(
                     "wrong comparison operator. Only '{}' and '{}' are supported for strings.",
```

Comment:
\+ and pointers

Reply:
I think I got confused. If we have a pointer to a struct and we want to use the value, would it match `ctx.load.is_ptr()` ?